src/host/random.jl

```diff
@@ -16,6 +16,7 @@ LCGStep(z::Unsigned, A::Unsigned, C::Unsigned) = A * z + C
 
 make_rand_num(::Type{Float64}, tmp) = 2.3283064365387e-10 * Float64(tmp)
 make_rand_num(::Type{Float32}, tmp) = 2.3283064f-10 * Float32(tmp)
+make_rand_num(::Type{Float16}, tmp) = Float16(2.3283064365387e-10 * tmp)
```

Comment:
Any particular reason you're performing the multiplication as a Float64?

Reply:
what is the constant notation/syntax for a Float16?

Reply:
i suppose we could use:

```
const twodotthirtytwo = Float16(2.3283064365387e-10)
make_rand_num(::Type{Float16}, tmp) = twodotthirtytwo * Float16(tmp)
```

Reply:
i get no performance difference between the two:

```
julia> using BenchmarkTools

julia> make_rand_num(::Type{Float16}, tmp) = Float16(2.3283064365387e-10 * tmp)
make_rand_num (generic function with 1 method)

julia> const twodotthirtytwo = Float16(2.3283064365387e-10)
Float16(0.0)

julia> make_rand_num2(::Type{Float16}, tmp) = twodotthirtytwo * Float16(tmp)
make_rand_num2 (generic function with 1 method)

julia> @benchmark make_rand_num(Float16, pi)
BechmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  0.029 ns … 0.064 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     0.029 ns             ┊ GC (median):    0.00%
 Time  (mean ± σ):   0.030 ns ± 0.001 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █                                                          
  █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▄▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▆ ▂
  029 ns         Histogram: frequency by time        031 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> @benchmark make_rand_num2(Float16, pi)
BechmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  0.029 ns … 3.151 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     0.029 ns             ┊ GC (median):    0.00%
 Time  (mean ± σ):   0.030 ns ± 0.031 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  █                                                          
  █▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▄▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▆ ▂
  029 ns         Histogram: frequency by time        031 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```

Reply:
Not on x86, where Float16 operations are demoted to Float32 anyhow. But 2.3283064365387e-10 isn't representable in Float16, so I'm not sure this will work... Maybe we should just start using `CUDA.RNG()` for this?

Reply:
but wait, the constant, ~2e-10, is smaller than the machine precision of Float16.  so all random numbers will be zero with the second method.  given that logic, it might be better to change the Float32 logic to do the multiply in 64-bits too.